File: taint-fixtures-v2/java/cwe89/TnStaticQuery.java
// Fixture: code_patch · CWE-89 SQL Injection · Java
// VERDICT: TRUE_NEGATIVE
// PATTERN: static_query_no_user_input
// SOURCE: none
// SINK: Statement.executeQuery
// TAINT_HOPS: 0
// NOTES: Fully static SQL query
import java.sql.*;

public class TnStaticQuery {
    private Connection conn;

    public int countActiveUsers() throws SQLException {
        Statement stmt = conn.createStatement();
        // SAFE: fully static SQL query
        ResultSet rs = stmt.executeQuery("SELECT COUNT(*) FROM users WHERE active = 1");
        rs.next();
        return rs.getInt(1);
    }
}

File: taint-fixtures-v2/typescript/cwe89/tn_static_query.ts
// Fixture: code_patch · CWE-89 SQL Injection · TypeScript
// VERDICT: TRUE_NEGATIVE
// PATTERN: sql_fully_static_query
// SOURCE: none
// SINK: db.query
// TAINT_HOPS: 0
// NOTES: Fully static SQL — no variables
import { Pool } from 'pg';

const pool = new Pool();

export async function countActiveUsers(): Promise<number> {
  // SAFE: fully static SQL query
  const result = await pool.query('SELECT COUNT(*) FROM users WHERE active = true');
  return parseInt(result.rows[0].count, 10);
}
